src/foundry/foundry.js/pointSources/visionSource.d.ts

```diff
@@ -0,0 +1,121 @@
+interface VisionSourceData extends PointSource.Data {
+  x: number;
+
+  y: number;
+
+  /** An optional z-index sorting for the source */
+  z: number;
+
+  /** The angle of rotation for this point source */
+  rotation: number;
+
+  /** The angle of emission for this point source */
+  angle: number;
+
+  /** The allowed radius of bright vision or illumination */
+  bright: number;
+
+  /** The allowed radius of dim vision or illumination */
+  dim: number;
+}
+
+/**
+ * A specialized subclass of the PointSource abstraction which is used to control the rendering of vision sources.
+ */
+declare class VisionSource extends PointSource {
+  /** @param object - The Token object that generates this vision source */
+  constructor(object: Token);
+
+  /**
+   * The current vision mesh for this source
+   * @defaultValue `this._createMesh(AdaptiveIlluminationShader)`
+   */
+  illumination: PIXI.Mesh;
+
+  /** @override */
+  static sourceType: 'vision';
+
+  /**
+   * Keys in the VisionSourceData structure which, when modified, change the appearance of the source
+   * @defaultValue `["dim", "bright"]`
+   * @internal
+   */
+  protected static _appearanceKeys: string[];
+
+  /**
+   * The object of data which configures how the source is rendered
+   * @defaultValue `{}`
+   */
+  data: Partial<VisionSourceData>;
+
+  /**
+   * The ratio of dim:bright as part of the source radius
+   * @defaultValue `undefined`
+   */
+  ratio: number | undefined;
+
+  /**
+   * The rendered field-of-vision texture for the source for use within shaders.
+   * @defaultValue `undefined`
+   */
+  fovTexture: PIXI.RenderTexture | undefined;
+
+  /**
+   * Track which uniforms need to be reset
+   * @defaultValue `{ illumination: true }`
+   * @internal
+   */
+  protected _resetUniforms: Record<string, boolean>;
+
+  /**
+   * To track if a source is temporarily shutdown to avoid glitches
+   * @defaultValue `{ illumination: false }`
+   * @internal
+   */
+  protected _shutdown: Record<string, boolean>;
+
+  /**
+   * Initialize the source with provided object data.
+   * @param data - Initial data provided to the point source
+   * @returns A reference to the initialized source
+   */
+  initialize(data?: Partial<VisionSourceData>): this;
+
+  fov: PIXI.Polygon | undefined;
```

Comment:
```suggestion
  /** @defaultValue `undefined` */
  fov: PIXI.Circle | undefined;
```

apparently, `PlaceableObject.fov` is documented / typed incorrectly, as the fov apparently always is a `PIXI.Circle` or `undefined`, not a `PIXI.Polygon`.

Reply:
I guess someone assumed that a `PIXI.Circle` subclasses `PIXI.Polygon`. Also I'd like to change it to `fov?: PIXI.Circle`, because it is never assigned `undefined`. Either it's there or not.